src/targets/rom.ts

```diff
@@ -0,0 +1,77 @@
+import ESPLoader from "../espLoader";
+
+export enum FLASH_SIZES {
+  "1MB" = 0x00,
+  "2MB" = 0x10,
+  "4MB" = 0x20,
+  "8MB" = 0x30,
+  "16MB" = 0x40,
+}
+
+export enum ESP8266_FLASH_SIZES {
+  "512KB" = 0x00,
+  "256KB" = 0x10,
+  "1MB" = 0x20,
+  "2MB" = 0x30,
+  "4MB" = 0x40,
+  "2MB-c1" = 0x50,
+  "4MB-c1" = 0x60,
+  "8MB" = 0x80,
+  "16MB" = 0x90,
+}
+
+export abstract class ROM {
+  // abstract read_efuse(loader: ESPLoader, offset: number): Promise<number>; //esp32
+
+  // abstract get_pkg_version(loader: ESPLoader): Promise<number>; // not in esp32s3
+
+  // abstract get_chip_revision(loader: ESPLoader): Promise<number>; esp32
+
+  abstract get_chip_description(loader: ESPLoader): Promise<string>;
+
+  abstract get_chip_features(loader: ESPLoader): Promise<string[]>;
+
+  abstract get_crystal_freq(loader: ESPLoader): Promise<number>;
+
+  abstract _d2h(d: number): string;
+
+  abstract read_mac(loader: ESPLoader): Promise<string>;
+
+  _post_connect?(loader: ESPLoader): Promise<void>;
+
+  get_erase_size(offset: number, size: number) {
+    return size;
+  }
+
+  FLASH_SIZES: { [key: string]: number } = {
+    "1MB": 0x00,
+    "2MB": 0x10,
+    "4MB": 0x20,
+    "8MB": 0x30,
+    "16MB": 0x40,
+  };
```

Comment:
This is a duplicate of above?

```suggestion
  FLASH_SIZES = FLASH_SIZES;
```